### PR TITLE
Remove analytics and play services dependency

### LINKED ISSUE #3: Provide pure-floss build flavor
The app now uses Google's proprietary play-services. Could you provide a completely free build flavor?

### PR DESCRIPTION
**Pull request by h4n23s**
Closes #3.

Hi again. 🙂

This PR implements two build flavors (``floss`` and ``play``).
``play`` makes use of google analytics (as before) while ``floss`` prints stack traces instead.

@IzzySoft Are these changes sufficient to re-enable builds on F-Droid?

### CODE REVIEW

**Comment by IzzySoft** [comment]
We'll see once our bots reported and we had an APK to check against. I cannot verify the code if that's what you ask, I'm not an Android dev. But I can give the `build.gradle` a look and make a crystal ball diagnosis of the possible results. And my crystal ball says the `build.gradle` looks promising :smiley:

**Comment by h4n23s** [comment]
Thats great, thank you! Since I don't know much about the F-Droid app verification and publication mechanisms, I was a bit worried that the GMS Gradle plugin would cause the app to be rejected, despite the fact that there's actually no tracking in the code. 🙂

**Comment by mattgmg1990** [comment]
This would be ok with me, but it's a little disappointing we have to duplicate all the code.

I actually have forked this project privately for my own play store version because I wanted more analytics to understand how people use the app and I added some additional functionality. So, I no longer would utilize the google analytics tracking code here personally. If we want to return this repository to full floss and just delete the analytics code and the gms dependency that would also be fine with me, or we can merge this, either way works.

**Comment by h4n23s** [comment]
Done! I just wasn't sure if you wanted to keep analytics, so I created a ``floss`` flavor as suggested by @IzzySoft. 🙂

**Comment by h4n23s** [comment]
It's ready! 🚀

**Comment by mattgmg1990** [comment]
Cool, yeah since I have a private fork I don't feel the need to keep them here.

Two change requests:
1. can we rename the PR to something like "remove analytics and play services dependency?"
2. You can just remove MiracastApplication if it will have no content and the reference in the manifest xml

**Comment by h4n23s** [comment]
Done.

**Comment by mattgmg1990** [comment]
cool thanks! will want to build and test it out but can't for a few hours until later, I'll merge after that.

File: app/src/main/java/com/mattgmg/miracastwidget/MainActivity.java
Comment (mattgmg1990) [inline app/src/main/java/com/mattgmg/miracastwidget/MainActivity.java:44]: We should either log to logcat with Log.e or just remove this. See https://stackoverflow.com/questions/3855187/is-it-a-bad-idea-to-use-printstacktrace-for-caugt-exceptions

Can you just remove the exception printing everywhere please? If we're catching and we don't need to catch for any reason other than to log, we can just remove the try/catch.

**Review by mattgmg1990 — CHANGES_REQUESTED** [review]
We should remove exception printing but otherwise looks good!

File: app/src/main/java/com/mattgmg/miracastwidget/MainActivity.java
Comment (h4n23s) [reply app/src/main/java/com/mattgmg/miracastwidget/MainActivity.java]: I'm sorry, I guess I just didn't notice using ``printStackTrace()`` at that moment.

> If we're catching and we don't need to catch for any reason other than to log, we can just remove the try/catch.

Maybe I've misunderstood you, but why would you want to remove ``try/catch`` when ``catch`` does nothing? Isn't that better than crashing the whole app, especially when some exceptions are device-dependent/only thrown on devices that don't support Miracast? Also, the method signature of ``onCreate()`` doesnt allow for throwing any non-runtime exceptions like ``PackageManager.NameNotFoundException``.

I've just removed exception logging for now. ``try/catch`` seems to serve another purpose besides preventing the app from crashing: When Miracast is not supported, a 🍞 is shown. 🙂

Comment (mattgmg1990) [reply app/src/main/java/com/mattgmg/miracastwidget/MainActivity.java]: Oh yes, of course. Sorry, I wasn't really thinking when I wrote that comment. You're right, we don't want to crash. Sure, just leaving it sounds good.

**Review by mattgmg1990 — APPROVED** [review]
looks good!

**Pull request merged**
